bolt-modules/boltlib/lib/puppet/functions/run_script.rb

```diff
@@ -108,7 +110,22 @@ def run_script_with_description(scope, script, targets, description = nil, optio
     # Send Analytics Report
     executor.report_function_call(self.class.name)
 
-    found = Puppet::Parser::Files.find_file(script, scope.compiler.environment)
+    # If the path is an absolute path, use that. Otherwise, commence the
+    # search.
+    found = if Pathname.new(script).absolute?
+              script
+            else
+              # Fall back to new-style paths if enabled.
+              fallback = executor.future&.fetch('file_paths', false)
+              # Split the module_name from the pattern used to find the file
+              module_name, file_pattern = Bolt::Util.split_file_path(script)
+              # Get the absolute path to the module root from Puppet
+              mod_path = scope.compiler.environment.module(module_name)&.path
+              # Search the module for the file, optionally falling back to new-style
+              # paths if enabled.
+              Bolt::Util.find_file(mod_path, file_pattern, fallback) if mod_path
+            end
```

Comment:
Could this be a method on `Bolt::Util` since it's used in multiple plan functions? I don't imagine this will need to change, but just in case it does that would reduce risk.

Reply:
Yeah, that's a good idea. I moved this to be on the executor, which feels more correct for passing scope around and will also let any other executors easily override this function.